### PR TITLE
Ruby SDK PoC

### PR DESCRIPTION
**Pull request by dgageot**
Giving a shot at the Ruby SDK because Ruby is beautiful!

+ [x] Hand craft a first version of the Container type. Just enough to get started
+ [ ] Replace the hand-crafted Container type with generated code
+ [ ] Implement more types
+ [ ] Sort out the best api for lazy eval of ids
+ [ ] Auto-provisionning?
+ [ ] Sort out the best structure for the ruby code
+ [x] Better README
+ [ ] Code samples on the website?

### CODE REVIEW

**Review by gerhard — COMMENTED** [review]
It's so good to see some Ruby in here - thank you for kicking this one off @dgageot!

I have added a few tiny comments. I'm sure that some of the typos were added to check if anyone is paying attention. I use that strategy when appropriate, it's a good one 😉

Would it be possible to add a task list in the description with the things that you think this PR needs before we can merge it? I am thinking something along the lines of `README.md`, tests (one integration is a good start), etc. The prefer small changes that flow often, which implies a smoothness to the method.

Last thing: anything that we can help with?

File: sdk/ruby/lib/dagger/client.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require "graphlient"
+
+module Dagger
+  class << self
+    def connect!
+      port = ENV["DAGGER_SESSION_PORT"]
+      token = ENV["DAGGER_SESSION_TOKEN"]
```
[the remark this answers is not in the document]
Comment (dgageot) [reply sdk/ruby/lib/dagger/client.rb]: This should be fixed now

File: sdk/ruby/lib/dagger.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+module Dagegr
```
[the remark this answers is not in the document]
Comment (dgageot) [reply sdk/ruby/lib/dagger.rb]: Not my first time!

File: sdk/ruby/lib/dagger/version.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+module Dagger
+    VERSION = "0.0.1"
+  end
```
[the remark this answers is not in the document]
Comment (dgageot) [reply sdk/ruby/lib/dagger/version.rb]: Fixed!

**Comment by dgageot** [comment]
> It's so good to see some Ruby in here - thank you for kicking this one off @dgageot!

Thank you @gerhard for reviewing.

> I have added a few tiny comments. I'm sure that some of the typos were added to check if anyone is paying attention. I use that strategy when appropriate, it's a good one 😉

Should be better now. It force pushed a new version of the code because I experimented with codegen and ended up refactoring quite a lot of things.

> Would it be possible to add a task list in the description with the things that you think this PR needs before we can merge it? I am thinking something along the lines of `README.md`, tests (one integration is a good start), etc. The prefer small changes that flow often, which implies a smoothness to the method.
> 
> Last thing: anything that we can help with?

**Comment by gerhard** [comment]
Thank you @dgageot for sticking with me - had a good few days off 🏝

I pushed a few changes, let me know what you think.

In the interest of merging this PR quicker, do you think that it would be worth limiting the outstanding scope to an integration test? To be honest, that is the only thing that we are missing before this looks good enough to merge. This is how we (+@shykes) think about pull requests: https://github.com/dagger/dagger/blob/v0.6.3/CONTRIBUTING.md#scope-of-pull-requests

My proposal would be to:
- update the task list in this PRs description to just one more outstanding item: `- [ ] Add integration test`
- add integration test (also remember to add it in our `test` GHA workflow)
- rebase the entire PR on top of `main`

When all checks pass, I would be more than happy to merge this 💪

When all is done, we look forward to more follow-ups. Maybe other Rubyists would like to help out? My proposal would be for you to come talk about it in one of our future Community Calls - https://dagger.io/events -  and see who else in our community would be interested to join this exciting new SDK.

WDYT?

**Comment by dgageot** [comment]
@gerhard Sorry, I need more free time! I'll try to take a look at the end of this week

**Comment by gerhard** [comment]
OK, no worries, will check again next week.

**Pull request closed without merging**